Support EasyRPG Savegame version chunk

Ghabry opened the pull request:
Filled with 600 to reflect 0.6.0 savegame type.

fmatthew5876 commented on src/version.h:
```diff
@@ -32,4 +32,16 @@
+ * 3rd digit: Patch
+ * 4rd digit: Usually 0. Increment only when having a good reason.
+ */
+#define PLAYER_SAVEGAME_VERSION 600
```
This should just be something like `PLAYER_MAJOR * 1000 + PLAYER_MINOR * 100 + PLAYER_PATCH * 10`

Then we don't need to update it separately every time and your comments become unnecessary.

Ghabry replied:
but we will not change the savegame format with every release.

fmatthew5876 replied:
We think we won't change it. That doesn't mean a save game bug can't show up in one particular release inadvertently. If we don mark every release, we wouldn't be able to tag this after the fact.

Ghabry replied:
hm actually this could be also interesting for statistics, because then we can see in the logs which player version was used before for creating the save.
I will think about a solution.

fmatthew5876 replied:
Yeah, that would be a great boon to debugging save game bug reports.

Ghabry commented:
added the discussed change, improved debug logging and added a warning when a save is loaded that is newer than the used EasyRPG Player version. (up to 9.9.9.9 supported, future proof ;))

This will emit a few saves with "540" until the 0.6 release PR is merged but shouldn't matter because that's only in continious builds.

fmatthew5876 commented on src/version.h:
```diff
@@ -32,4 +32,10 @@
+ * Version written to the easyrpg_data.version field in savegames.
+ * Increment the last digit (0) only when having a good reason.
+ */
+#define PLAYER_SAVEGAME_VERSION PLAYER_MAJOR * 1000 + PLAYER_MINOR * 100 + PLAYER_PATCH * 10 + 0
```
Put `()` around the expression. Otherwise someone could get a surprise when doing something like `PLAYER_SAVEGAME_VERSION * 10`

fmatthew5876 commented on src/player.cpp:
```diff
@@ -825,12 +827,37 @@ static void OnMapSaveFileReady(FileRequestResult*) {
+	} else {
+		verstr << "EasyRPG Player ";
+		char verbuf[6];
+		sprintf(verbuf, "%d.%d.%d", ver / 1000 % 10, ver / 100 % 10, ver / 10 % 10);
```
Use `snprintf()` for extra safety

Ghabry replied:
I wanted to but I somehow expect that at least one platform port has a broken stdlib, well I will change it an dwe will see what happens.

fmatthew5876 replied:
Also I would just make the buffer large like 64 bytes. What if we have `10.0.0` or something bigger one day? Are you going to remember to resize the buffer? I know I wouldn't.

fmatthew5876 replied:
I see, if you make it bigger (like 32 or more), `sprintf()` is probably fine.

Ghabry commented:
Jenkins: test this please

fmatthew5876 commented:
looks good. Great feature! Thanks for getting this in before release.
